marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction.hs

```diff
@@ -222,3 +235,21 @@ renderLoadMarloweContextSelectorOTel = \case
             , Just ("marlowe.payout_script_hash", fromString $ show payoutScriptHash)
             ]
       }
+
+mkCommandLineRoleTokenMintingPolicy :: (MonadUnliftIO m, MonadFail m) => String -> MkRoleTokenMintingPolicy m
+mkCommandLineRoleTokenMintingPolicy cmd TxOutRef{..} roleTokens = do
+  let txIdArg = filter (/= '"') $ show txId
+  let txIxArg = show $ unTxIx txIx
+  let roleTokenArgs = do
+        (TokenName name, quantity) <- Map.toList roleTokens
+        [T.unpack $ encodeBase16 name, show quantity]
+  let args = txIdArg : txIxArg : roleTokenArgs
+  bracket
+    (createProcess (P.proc cmd args){std_out = CreatePipe, std_err = CreatePipe})
```

Comment:
I expected that it would have been more straightforward to  pass the minting information as a JSON object on `stdin` instead of as command-line arguments.

Reply:
That's another option, I can rework this to support that.